Zstandard compiles and run on m68k cpus

Cyan4973 opened the pull request:
fix #2756

Cyan4973 committed:
added alignment test

and fix an incorrect alignment check in cwksp which was failing on m68k

Cyan4973 committed:
fix align conditions for huf_compress

Cyan4973 committed:
m68k CI tests on GA are now compulsory

terrelln commented on lib/common/mem.h:
```diff
@@ -80,6 +80,26 @@ extern "C" {
+
+#else
+
+# define MEM_ALIGN_COND(T) 1  /* most likely incorrect, just to pass tests */
```
Won't this break some of the alignment in `huf_compress.c` on non-supported compilers?

Could we maybe default to `MIN(roundUpToPow2(sizeof(T)), sizeof(size_t))`?

Cyan4973 committed:
move the alignment macro to compiler.h

because mem.h is dropped in the Linux kernel.

Changed macro definition order (gcc/clang/msvc before c11)
due to a limitation in the kernel source builder.

Changed the backup to sizeof(),
reverting to previous behavior when no support of alignof() is detected.

Cyan4973 committed:
bound alignment backup to sizeof(void*)

Cyan4973 committed:
changed macro name to ZSTD_ALIGNOF

for better consistency
